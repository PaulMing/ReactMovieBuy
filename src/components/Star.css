.star {
  display: inline-block;
  background: #fff url("../icons/star-o.svg") 0 50%/contain repeat-x; }
  .star__top {
    height: 100%;
    background: #fff url("../icons/star.svg") 0 50%/contain repeat-x; }
