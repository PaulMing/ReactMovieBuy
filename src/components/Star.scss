.star {
  display: inline-block;
  background: #fff url("../icons/star-o.svg") 0 50% / contain repeat-x;

  &__top {
    height: 100%;
    background: #fff url("../icons/star.svg") 0 50% / contain repeat-x;
  }
}
