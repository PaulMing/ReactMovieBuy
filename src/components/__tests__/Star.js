import React from 'react';
import Star from '../Star';
import renderer from 'react-test-renderer';

// it;//断言 -> if xxx 其有xx行为 
it('渲染正常', () => {
  const tree = renderer.create(<Star value={8} />).toJSON();
  expect(tree).toMatchSnapshot();
});
