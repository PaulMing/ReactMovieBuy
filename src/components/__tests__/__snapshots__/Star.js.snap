// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染正常 1`] = `
<div
  className="star"
  style={
    Object {
      "backgroundSize": "15px",
      "height": 15,
      "width": 75,
    }
  }
>
  <div
    className="star__top"
    style={
      Object {
        "backgroundSize": "15px",
        "width": "80%",
      }
    }
  />
</div>
`;
